client/src/main/java/org/opensearch/geospatial/action/IpEnrichmentActionClient.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.action;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.action.ActionFuture;
+import org.opensearch.core.action.ActionResponse;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Facade to provide GeoLocation enrichment for other plugins.
+ */
+@Log4j2
+@AllArgsConstructor
+public class IpEnrichmentActionClient {
+
+    NodeClient nodeClient;
```

Comment:
any reason for NodeClient being package private?

Reply:
Nope, updated with `final private` :)